src/prices/coingecko.ts

```diff
@@ -0,0 +1,53 @@
+import BigNumber from 'bignumber.js'
+import { logger } from '../logger'
+import fetch from 'cross-fetch'
+
+export const BASE_URL =
+  'https://api.coingecko.com/api/v3/simple/token_price/celo'
+
+const USD = 'usd'
+
+/**
+ * Returns token prices compared to the base token.
+ */
+export async function fetchPrices(tokenAddresses: string[], baseToken: string) {
+  try {
+    const url = `${BASE_URL}?contract_addresses=${tokenAddresses.join(
+      ',',
+    )}&vs_currencies=usd`
+    logger.debug(`Fetching prices with ${url}`)
+    const response = await fetch(url)
+    const jsonResponse = await response.json()
+    logger.debug(`Response from coingecko: ${JSON.stringify(jsonResponse)}`)
```

Comment:
Easier to browser in Logs Explorer:
```suggestion
    logger.debug({ jsonResponse }, `Response from coingecko`)
```